hedera-node/src/jmh/java/com/hedera/services/ledger/StakingActivityModule.java

```diff
@@ -0,0 +1,162 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.ledger;
+
+import static com.hedera.services.mocks.MockDynamicProperties.mockPropertiesWith;
+
+import com.hedera.services.config.AccountNumbers;
+import com.hedera.services.context.SideEffectsTracker;
+import com.hedera.services.context.TransactionContext;
+import com.hedera.services.context.annotations.CompositeProps;
+import com.hedera.services.context.properties.GlobalDynamicProperties;
+import com.hedera.services.context.properties.PropertySource;
+import com.hedera.services.context.properties.SupplierMapPropertySource;
+import com.hedera.services.ledger.accounts.staking.RewardCalculator;
+import com.hedera.services.ledger.accounts.staking.StakeChangeManager;
+import com.hedera.services.ledger.accounts.staking.StakeInfoManager;
+import com.hedera.services.ledger.accounts.staking.StakePeriodManager;
+import com.hedera.services.ledger.backing.BackingAccounts;
+import com.hedera.services.ledger.backing.BackingStore;
+import com.hedera.services.ledger.interceptors.StakingAccountsCommitInterceptor;
+import com.hedera.services.ledger.properties.AccountProperty;
+import com.hedera.services.ledger.properties.ChangeSummaryManager;
+import com.hedera.services.mocks.MockAccountNumbers;
+import com.hedera.services.mocks.MockAccountTracking;
+import com.hedera.services.mocks.MockEntityCreator;
+import com.hedera.services.mocks.MockProps;
+import com.hedera.services.mocks.MockRecordsHistorian;
+import com.hedera.services.mocks.MockTransactionContext;
+import com.hedera.services.records.RecordsHistorian;
+import com.hedera.services.setup.InfrastructureBundle;
+import com.hedera.services.setup.InfrastructureType;
+import com.hedera.services.state.EntityCreator;
+import com.hedera.services.state.merkle.MerkleAccount;
+import com.hedera.services.state.merkle.MerkleNetworkContext;
+import com.hedera.services.state.merkle.MerkleStakingInfo;
+import com.hedera.services.state.validation.AccountUsageTracking;
+import com.hedera.services.utils.EntityNum;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.swirlds.merkle.map.MerkleMap;
+import dagger.Binds;
+import dagger.Module;
+import dagger.Provides;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+import javax.inject.Singleton;
+
+@Module
+public interface StakingActivityModule {
+    @Binds
+    @Singleton
+    EntityCreator bindEntityCreator(MockEntityCreator entityCreator);
+
+    @Binds
+    @Singleton
+    RecordsHistorian bindRecordsHistorian(MockRecordsHistorian recordsHistorian);
+
+    @Binds
+    @Singleton
+    TransactionContext TransactionContext(MockTransactionContext transactionContext);
```

Comment:
nit : naming for function `transactionContext`

Reply:
Ah typo, will fix!